### PR TITLE
chore: update to latest Nx 11 release

### PR DESCRIPTION
**Pull request by brandonroberts**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x]  The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Packages are using the Nx 10.x release packages

Closes #

## What is the new behavior?

Packages are using the latest Nx 11.x release packages

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information

Did have to remove some of the `@angular-devkit/core` references, but this is not a breaking change, as version 11 of NgRx should be used with version 11 packages of Angular/CLI/Devkit.

### CODE REVIEW

**Comment by timdeschryver** [comment]
I'll review this later today, tomorrow at the latest.

File: migrations.json [hunk 1]
```diff
@@ -0,0 +1,249 @@
+{
```
Comment (timdeschryver) [inline migrations.json:1]: I'm not familiar with this, but I assume NX keeps track which migrations need to run based on this list, that's why we include it to our repo?

File: modules/schematics/jest.config.js [hunk 2]
```diff
@@ -1,7 +1,6 @@
 module.exports = {
-  name: 'schematics',
-  displayName: 'Schematics',
-  preset: '../../jest.config.js',
+  displayName: 'schematics',
```
Comment (timdeschryver) [inline modules/schematics/jest.config.js:2]: 🙈
```suggestion
  displayName: 'Schematics',
```

**Review by timdeschryver — APPROVED** [review]

File: migrations.json (hunk 1 above)
Comment (brandonroberts) [reply migrations.json:1]: Yes, it runs the migrations based on the version from/to you're using. We probably don't need to keep it though

**Pull request merged**